src/shared/infrastructure/tracking/faro/faro.ts

```diff
@@ -0,0 +1,58 @@
+import { Faro, getWebInstrumentations, initializeFaro } from '@grafana/faro-web-sdk';
+import { TracingInstrumentation } from '@grafana/faro-web-tracing';
+import { config } from '@grafana/runtime';
+
+import { PYROSCOPE_APP_ID } from '../../../../constants';
+import { GIT_COMMIT } from '../../../../version';
+import { getFaroEnvironment } from './getFaroEnvironment';
+
+let faro: Faro | null = null;
+
+export const getFaro = () => faro;
+export const setFaro = (instance: Faro | null) => (faro = instance);
+
+export function initFaro() {
+  if (getFaro()) {
+    return;
+  }
+
+  const faroEnvironment = getFaroEnvironment();
+  if (!faroEnvironment) {
+    return;
+  }
+
+  const { environment, faroUrl, appName } = faroEnvironment;
+
+  const { apps, bootData } = config;
+  const appRelease = apps[PYROSCOPE_APP_ID].version;
+  const userEmail = bootData.user.email;
+
+  setFaro(
+    initializeFaro({
+      url: faroUrl,
+      app: {
+        name: appName,
+        release: appRelease,
+        version: GIT_COMMIT,
+        environment,
+      },
+      user: {
+        email: userEmail,
+      },
+      instrumentations: [
+        ...getWebInstrumentations({
+          captureConsole: false,
+        }),
+        new TracingInstrumentation(),
+      ],
+      isolate: true,
+      beforeSend: (event) => {
```

Comment:
BTW. just ouf of curiosity - Is this needed? Isn't Faro backend/app configured to accept logs from specific routes? 🤔  What is preventing anyone else from pushing to the same collector some rubbish data? 🤔